### PR TITLE
Fix snappy version

### PR DESCRIPTION
**Pull request by jonashaag**
Fix for https://github.com/pandas-dev/pandas/pull/46260

- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by jonashaag** [comment]
Should we also add `python-snappy` to one of the test envs?

**Comment by mroeschke** [comment]
> Should we also add `python-snappy` to one of the CI envs?

Yes should be added to:

actions-38.yaml
actions-38-downstream_compat.yaml
actions-38-minimum_versions.yaml  (pick latest as a pin)
actions-39.yaml
actions-310.yaml


And noted `install.rst` (under compression section)

File: pandas/compat/_optional.py [hunk 1]
```diff
@@ -72,6 +73,10 @@ def get_version(module: types.ModuleType) -> str:
         if module.__name__ == "brotli":
             # brotli doesn't contain attributes to confirm it's version
             return ""
+        if module.__name__ == "snappy":
+            # snappy doesn't contain attributes to confirm it's version
```
Comment (jbrockmendel) [inline pandas/compat/_optional.py:77]: `snappy.__version__` for me says "3.0.3"

Comment (jonashaag) [reply pandas/compat/_optional.py:77]: Are you confusing it with https://snappy.math.uic.edu/?

Comment (jbrockmendel) [reply pandas/compat/_optional.py:77]: Yes, good catch!

**Review by mroeschke — APPROVED** [review]
Awesome, thanks for alphabetizing too. LGTM

**Comment by jreback** [comment]
thanks @jonashaag

**Pull request merged**